otj-server-core/src/main/java/com/opentable/server/PortSelector.java

```diff
@@ -92,80 +101,62 @@ public PortSelector(Environment environment) {
+        return PortSelection.empty(name);
     }
 
-    public PortSelection getWithDefault(final String springProperty, final int ordinal, final String namedPort, int defaultV) {
-        final PortSelection portSelection = get(springProperty, ordinal, namedPort );
-        return portSelection.hasValue() ? portSelection : new PortSelection(String.valueOf(defaultV), PortSource.FROM_DEFAULT_VALUE, String.valueOf(defaultV));
+    public PortSelection getWithDefault(final String springProperty, final String namedPort, int defaultV) {
+        final PortSelection portSelection = get(springProperty, namedPort);
+        return portSelection.hasValue() ? portSelection : new PortSelection(springProperty, String.valueOf(defaultV), PortSource.FROM_DEFAULT_VALUE, String.valueOf(defaultV));
     }
 
-    PortSelection getJMXPort() {
-        return getWithDefault(
-                JMX_PORT,
-                 1,
-                 "PORT_JMX",
-                0);
+    private PortSelection getJMXPort() {
+        return getWithDefault(JMX_PORT, "PORT_JMX", 0);
     }
 
-    PortSelection getActuatorPort() {
-        return getWithDefault(MANAGEMENT_SERVER_PORT,
-                2,
-                "PORT_ACTUATOR",0);
+    private PortSelection getActuatorPort() {
+        return get(MANAGEMENT_SERVER_PORT, "PORT_ACTUATOR");
     }
 
     public Map<String, PortSelection> getPortSelectionMap() {
-        final Map<String, PortSelector.PortSelection> portSelectionMap = new HashMap<>();
-
-        // server.port
-        //TODO: won't this conflict with the default-http? Discuss with Dmitry.
-        //TODO: Dmitry, the docs I found claimed ssl is on by default? and keystore is what in combo activates it
-        String namedPort = Boolean.parseBoolean(
-                environment.getProperty(SERVER_SSL_ENABLED, "true")) && environment.getProperty("server.ssl.key-store") != null ? "PORT_HTTPS" : "PORT_HTTP";
-        //TODO: discuss with dmitry: shouldn't the default be null? eg not flipping this on at all
-        portSelectionMap.put(SERVER_PORT, getWithDefault(SERVER_PORT, 0, namedPort, 8080 ));
-
-        // ot.httpserver.connector.default-http.port
-        namedPort = "http".equalsIgnoreCase(
-                environment.getProperty(HTTPSERVER_CONNECTOR_DEFAULT_HTTP_PROTOCOL, "http")) ? "PORT_HTTP"
-                : "PORT_HTTPS";
-        portSelectionMap.put(HTTPSERVER_CONNECTOR_DEFAULT_HTTP_PORT,
-                getWithDefault(HTTPSERVER_CONNECTOR_DEFAULT_HTTP_PORT, 0, namedPort, -1));
-
-        // jmx
-        portSelectionMap.put(JMX_PORT, getJMXPort());
-        // actuator
-        //TODO: discuss with dmitry: shouldn't the default be null? eg not flipping this on at all
-        portSelectionMap.put(MANAGEMENT_SERVER_PORT, getActuatorPort());
-
-        return portSelectionMap;
+        Map<String, PortSelection> res = Arrays.stream(environment.getProperty("ot.httpserver.active-connectors", "default-http").split(","))
```

Comment:
scan active connectors and build full list

Reply:
trim whitespace. Whitespace is mean.